### ISSUE TITLE
Input and forms: make the 'required' indicator actual text instead of just the asterisk

### ISSUE THREAD

**Issue by elisa-est** [Energy-Saving-Trust/Design-System#26]
Per: https://www.w3.org/WAI/test-evaluate/easy-checks/required-fields/#:~:text=The%20best%20way%20to%20indicate,users%20because%20it%27s%20considered%20punctuation.

"Many forms use a red asterisk, ‘*’, in the label. The asterisk is often not announced to screen reader users because it’s considered punctuation. People with low vision can often miss the ‘*’ too due to it’s small size and the color that is used."

We want to use actual text written out saying 'required' instead of just the asterisk to indicate required fields in:
1. Input
2. Select
3. Dropdown
4. Any other form field components

@EST-Karol and @lynsiepatient fyi, I added this issue.
